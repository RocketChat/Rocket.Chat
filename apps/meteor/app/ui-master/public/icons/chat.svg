<svg id="chat" viewBox="0 0 20 20">
  <path d="M3.36312881,15.7593532 C3.32888874,15.7771726 3.29959994,15.7925621 3.27370934,15.8064174 C3.73031444,16.1917386 4.92393678,16.4549204 7.10274479,16.4476544 L7.21881505,16.4476539 C8.55574002,16.4476543 9.6998097,16.021859 10.6510241,15.170268 L12.188241,15.2712054 C11.5513943,15.8887667 11.2142468,16.2130071 11.1767984,16.2439265 C10.1491488,17.09241 8.76567843,17.5637907 7.21881505,17.5637907 L7.10806092,17.5644215 L6.97721254,17.5644231 C4.30241208,17.5644231 2.62246666,17.1951018 2.10341288,16.0930165 C1.82161029,15.4951149 2.13151102,15.1455492 2.84114675,14.7722012 C3.31318797,14.5227311 3.32172914,14.5103228 3.24657386,14.3473099 C3.20302404,14.2436958 3.16526615,14.1604157 3.07972552,13.9763174 C2.74839464,13.2684751 2.62592155,12.9023078 2.62592155,12.4061039 C2.62592155,11.562445 2.83045155,10.7486059 3.21214892,10.0211892 C3.42204019,9.66099218 3.78721597,9.06536542 4.270822,8.82818513 C4.270822,8.80275057 4.53792399,10.0211892 4.53792399,10 C4.45304843,10.1183326 4.36401443,10.2496015 4.270822,10.3938065 C3.78721597,11.0991197 3.7445882,11.6631394 3.7445882,12.4061039 C3.7445882,12.6985231 3.82924269,12.9394008 4.09330148,13.5047226 C4.19543836,13.7251753 4.22995648,13.801243 4.270822,13.8994264 C4.58782821,14.5810467 4.36137367,15.1036933 3.78721597,15.5080098 C3.65916127,15.5981847 3.55361796,15.6588423 3.36312881,15.7593532 Z M16.5696621,12.3226926 C16.5350205,12.3039671 16.4951623,12.2829692 16.4468239,12.2578135 C16.2265134,12.1415693 16.1048536,12.0716493 15.9574055,11.9678178 C15.2986909,11.5039571 15.0399564,10.9068098 15.4049351,10.1218159 C15.4504792,10.0121781 15.4903754,9.92425877 15.608285,9.66976084 C15.914896,9.01333804 16.0132801,8.73339397 16.0132801,8.39198996 C16.0132801,5.80517045 13.90418,3.70028722 11.3120655,3.70028722 C8.71995111,3.70028722 6.61085101,5.80517045 6.61085101,8.39198996 C6.61085101,11.1096858 8.86296092,13.082961 11.9786225,13.082961 L12.112848,13.0829616 C14.6633683,13.0914671 16.0501708,12.7797435 16.5696621,12.3226926 Z M16.5612822,10.6280948 C16.4685253,10.8294269 16.4833349,10.8509584 17.0358533,11.1429611 C17.8499187,11.5712649 18.2032866,11.969896 17.8824254,12.6506683 C17.2857779,13.9175051 15.3458564,14.3439799 12.2581228,14.3439799 L12.1073975,14.3439799 L11.9780624,14.3432467 C8.14202747,14.3432482 5.34836876,11.8401459 5.34836876,8.39198996 C5.34836876,5.10955211 8.0234498,2.44 11.3120655,2.44 C14.6008168,2.44 17.276494,5.10968724 17.276494,8.39198996 C17.276494,8.96303774 17.1353362,9.38506819 16.7528285,10.2022436 C16.6539056,10.4151439 16.6101724,10.5116034 16.5612822,10.6280948 Z"/>
</svg>