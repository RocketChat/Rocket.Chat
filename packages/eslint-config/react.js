/** @type {import('eslint').ESLint.ConfigData} */
const config = {
	plugins: ['react', 'react-hooks', 'jsx-a11y'],
	extends: ['plugin:jsx-a11y/recommended'],
	rules: {
		'react-hooks/exhaustive-deps': 'error',
		'react-hooks/rules-of-hooks': 'error',
		'react/display-name': 'error',
		'react/jsx-curly-brace-presence': 'error',
		'react/jsx-fragments': ['error', 'syntax'],
		'react/jsx-key': ['error', { checkFragmentShorthand: true, checkKeyMustBeforeSpread: true, warnOnDuplicates: true }],
		'react/jsx-no-undef': 'error',
		'react/jsx-uses-react': 'error',
		'react/jsx-uses-vars': 'error',
		'react/no-multi-comp': 'error',
		'jsx-a11y/no-autofocus': [2, { ignoreNonDOM: true }],
	},
	settings: {
		react: {
			version: 'detect',
		},
	},
	overrides: [
		{
			files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
			rules: {
				'react/display-name': 'off',
				'react/no-multi-comp': 'off',
			},
		},
	],
};

module.exports = config;
