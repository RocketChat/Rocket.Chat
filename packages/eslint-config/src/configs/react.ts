import jsxA11y from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';
import reactRefresh from 'eslint-plugin-react-refresh';
import testingLibrary from 'eslint-plugin-testing-library';

import globals from 'globals';
import type { Linter } from 'eslint';

const configs = {
	'testing-library': {
		files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
		plugins: {
			'testing-library': testingLibrary,
		},
		rules: {
			'testing-library/await-async-events': ['warn', { eventModule: 'userEvent' }],
			'testing-library/await-async-queries': 'warn',
			'testing-library/await-async-utils': 'warn',
			'testing-library/no-await-sync-events': ['warn', { eventModules: ['fire-event'] }],
			'testing-library/no-await-sync-queries': 'warn',
			'testing-library/no-container': 'warn',
			'testing-library/no-debugging-utils': 'warn',
			'testing-library/no-dom-import': ['warn', 'react'],
			'testing-library/no-global-regexp-flag-in-query': 'warn',
			'testing-library/no-manual-cleanup': 'warn',
			'testing-library/no-node-access': 'warn',
			'testing-library/no-promise-in-fire-event': 'warn',
			'testing-library/no-render-in-lifecycle': 'warn',
			'testing-library/no-unnecessary-act': 'warn',
			'testing-library/no-wait-for-multiple-assertions': 'warn',
			'testing-library/no-wait-for-side-effects': 'warn',
			'testing-library/no-wait-for-snapshot': 'warn',
			'testing-library/prefer-find-by': 'warn',
			'testing-library/prefer-presence-queries': 'warn',
			'testing-library/prefer-query-by-disappearance': 'warn',
			'testing-library/prefer-screen-queries': 'warn',
			'testing-library/render-result-naming-convention': 'warn',
		},
	},
	'jsx-a11y': {
		plugins: {
			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
			'jsx-a11y': jsxA11y,
		},
		rules: {
			'jsx-a11y/click-events-have-key-events': 'warn',
			'jsx-a11y/no-static-element-interactions': 'warn',
			'jsx-a11y/no-autofocus': 'warn',
			'jsx-a11y/media-has-caption': 'warn',
			'jsx-a11y/iframe-has-title': 'warn',
			'jsx-a11y/alt-text': 'warn',
		},
		settings: {
			react: {
				version: 'detect',
			},
		},
	},
} satisfies Record<string, Linter.Config>;

type ReactRules = {
	[Key in keyof (typeof reactPlugin)['rules'] as `react/${Key}`]?: Linter.RuleEntry<unknown[]>;
};

export default function react(config: Linter.Config<ReactRules> = {}): Linter.Config[] {
	return [
		reactPlugin.configs.flat['recommended']!,
		reactPlugin.configs.flat['jsx-runtime']!,
		reactHooks.configs['recommended-latest'],
		reactRefresh.configs.recommended,
		configs['jsx-a11y'],
		configs['testing-library'],
		{
			languageOptions: {
				globals: globals.browser,
			},
		},
		config,
	];
}
