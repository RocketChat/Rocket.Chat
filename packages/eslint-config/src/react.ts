import tseslint, { type ConfigWithExtends } from 'typescript-eslint';
import jsxA11y from 'eslint-plugin-jsx-a11y';
import reactPlugin from 'eslint-plugin-react';
import reactHooks from 'eslint-plugin-react-hooks';
import reactRefresh from 'eslint-plugin-react-refresh';
import testingLibrary from 'eslint-plugin-testing-library';
import storybook from 'eslint-plugin-storybook';
import globals from 'globals';
import type { FlatConfig } from '@typescript-eslint/utils/ts-eslint';

const configs = {
	'testing-library': {
		files: ['**/*.stories.js', '**/*.stories.jsx', '**/*.stories.ts', '**/*.stories.tsx', '**/*.spec.tsx'],
		plugins: {
			'testing-library': testingLibrary,
		},
		rules: {
			'testing-library/await-async-events': ['warn', { eventModule: 'userEvent' }],
			'testing-library/await-async-queries': 'warn',
			'testing-library/await-async-utils': 'warn',
			'testing-library/no-await-sync-events': ['warn', { eventModules: ['fire-event'] }],
			'testing-library/no-await-sync-queries': 'warn',
			'testing-library/no-container': 'warn',
			'testing-library/no-debugging-utils': 'warn',
			'testing-library/no-dom-import': ['warn', 'react'],
			'testing-library/no-global-regexp-flag-in-query': 'warn',
			'testing-library/no-manual-cleanup': 'warn',
			'testing-library/no-node-access': 'warn',
			'testing-library/no-promise-in-fire-event': 'warn',
			'testing-library/no-render-in-lifecycle': 'warn',
			'testing-library/no-unnecessary-act': 'warn',
			'testing-library/no-wait-for-multiple-assertions': 'warn',
			'testing-library/no-wait-for-side-effects': 'warn',
			'testing-library/no-wait-for-snapshot': 'warn',
			'testing-library/prefer-find-by': 'warn',
			'testing-library/prefer-presence-queries': 'warn',
			'testing-library/prefer-query-by-disappearance': 'warn',
			'testing-library/prefer-screen-queries': 'warn',
			'testing-library/render-result-naming-convention': 'warn',
		},
	},
	'jsx-a11y': {
		plugins: {
			'jsx-a11y': jsxA11y,
		},
		rules: {
			'jsx-a11y/click-events-have-key-events': 'warn',
			'jsx-a11y/no-static-element-interactions': 'warn',
			'jsx-a11y/no-autofocus': 'warn',
			'jsx-a11y/media-has-caption': 'warn',
			'jsx-a11y/iframe-has-title': 'warn',
			'jsx-a11y/alt-text': 'warn',
		},
		settings: {
			react: {
				version: 'detect',
			},
		},
	},
} satisfies Record<string, ConfigWithExtends>;

function react({ languageOptions, ...config }: ConfigWithExtends = {}): FlatConfig.ConfigArray {
	return tseslint.config(
		{
			plugins: {
				react: reactPlugin,
			},
			languageOptions: {
				parserOptions: {
					ecmaFeatures: {
						jsx: true,
					},
					jsxPragma: null,
				},
			},
			rules: {
				...reactPlugin.configs.flat.recommended.rules,
				...reactPlugin.configs.flat['jsx-runtime'].rules,
				'react/no-unescaped-entities': 'warn',
				'react/no-children-prop': 'warn',
				'react/prop-types': 'warn',
				'react/display-name': 'warn',
				'react/jsx-curly-brace-presence': 'warn',
				'react/jsx-fragments': ['warn', 'syntax'],
				'react/jsx-key': ['warn', { checkFragmentShorthand: true, checkKeyMustBeforeSpread: true, warnOnDuplicates: true }],
				'react/jsx-no-undef': 'warn',
				'react/jsx-uses-vars': 'warn',
				'react/jsx-no-target-blank': 'warn',
				'react/no-multi-comp': 'warn',
				'react/no-direct-mutation-state': 'warn',
				'react/no-unknown-property': 'warn',
			},
		},
		reactHooks.configs['recommended-latest'],
		{
			rules: {
				'react-hooks/exhaustive-deps': 'warn',
				'react-hooks/rules-of-hooks': 'warn',
			},
		},
		reactRefresh.configs.recommended,
		{
			rules: {
				'react-refresh/only-export-components': 'warn',
			},
		},
		storybook.configs['flat/recommended'],
		{
			rules: {
				'storybook/no-renderer-packages': 'warn',
			},
		},
		configs['jsx-a11y'],
		configs['testing-library'],
		{
			languageOptions: {
				globals: globals.browser,
				...languageOptions,
			},
		},
		config,
	);
}

export default react;
