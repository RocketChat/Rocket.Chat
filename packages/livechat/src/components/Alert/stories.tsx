import { action } from '@storybook/addon-actions';
import type { Meta, StoryFn } from '@storybook/preact';
import type { ComponentProps } from 'preact';

import Alert from '.';
import { loremIpsum, screenDecorator } from '../../../.storybook/helpers';

export default {
	title: 'Components/Alert',
	component: Alert,
	args: {
		success: false,
		warning: false,
		error: false,
		color: '',
		timeout: 5000,
		onDismiss: action('dismiss'),
		children: loremIpsum({ count: 3, units: 'words' }),
	},
	parameters: {
		layout: 'centered',
	},
	decorators: [screenDecorator],
} satisfies Meta<ComponentProps<typeof Alert>>;

const Template: StoryFn<ComponentProps<typeof Alert>> = (args) => <Alert {...args} />;

export const Default = Template.bind({});
Default.storyName = 'default';
Default.args = {};

export const Success = Template.bind({});
Success.storyName = 'success';
Success.args = {
	success: true,
};

export const Warning = Template.bind({});
Warning.storyName = 'warning';
Warning.args = {
	warning: true,
};

export const Error = Template.bind({});
Error.storyName = 'error';
Error.args = {
	error: true,
};

export const CustomColor = Template.bind({});
CustomColor.storyName = 'custom color';
CustomColor.args = {
	color: '#175CC4',
};

export const WithLongTextContent = Template.bind({});
WithLongTextContent.storyName = 'with long text content';
WithLongTextContent.args = {
	children: loremIpsum({ count: 30, units: 'words' }),
};

export const WithoutTimeout = Template.bind({});
WithoutTimeout.storyName = 'without timeout';
WithoutTimeout.args = {
	timeout: 0,
};
