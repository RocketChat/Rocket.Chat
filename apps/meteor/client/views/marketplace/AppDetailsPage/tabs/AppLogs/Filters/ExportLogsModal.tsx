import {
	Box,
	Button,
	Field,
	FieldLabel,
	FieldRow,
	Label,
	Modal,
	ModalClose,
	ModalContent,
	ModalFooter,
	ModalFooterControllers,
	ModalHeader,
	ModalTitle,
	NumberInput,
	RadioButton,
} from '@rocket.chat/fuselage';
import { useRouteParameter } from '@rocket.chat/ui-contexts';
import { Controller, useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';

import type { AppLogsFilterFormData } from '../useAppLogsFilterForm';

type ExportLogsModalProps = {
	onClose: () => void;
	filterValues: AppLogsFilterFormData;
	onConfirm: (url: string) => void;
};

type FormDataType = {
	type: 'json' | 'csv';
	count: 'max' | 'custom';
	customExportAmount: number;
};

export const ExportLogsModal = ({ onClose, filterValues, onConfirm }: ExportLogsModalProps) => {
	const { t } = useTranslation();

	const appId = useRouteParameter('id');

	const {
		control,
		watch,
		getValues,
		formState: { isValid },
	} = useForm<FormDataType>({
		defaultValues: {
			type: 'json',
			count: 'max',
			customExportAmount: 100,
		},
	});

	const type = watch('type');
	const count = watch('count');

	const getFileUrl = ({
		severity,
		event,
		instance,
		startDate,
		endDate,
		count,
		type,
		startTime,
		endTime,
	}: AppLogsFilterFormData & { type: 'json' | 'csv'; count: number }): string => {
		let baseUrl = `/api/apps/${appId}/export-logs?`;
		if (severity && severity !== 'all') {
			baseUrl += `logLevel=${severity}&`;
		}
		if (event && event !== 'all') {
			baseUrl += `method=${event}&`;
		}
		if (instance && instance !== 'all') {
			baseUrl += `instanceId=${instance}&`;
		}
		if (startDate) {
			baseUrl += `startDate=${new Date(`${startDate}T${startTime}`).toISOString()}&`;
		}
		if (endDate) {
			baseUrl += `endDate=${new Date(`${endDate}T${endTime}`).toISOString()}&`;
		}
		if (count) {
			baseUrl += `count=${count}&`;
		}
		baseUrl += `type=${type}`;
		return baseUrl;
	};

	const handleConfirm = (): void => {
		const url = getFileUrl({ ...filterValues, type, count: count === 'max' ? 2000 : getValues('customExportAmount') });
		onConfirm(url);
		onClose();
	};

	return (
		<Modal>
			<ModalHeader>
				<ModalTitle>{t('Export')}</ModalTitle>
				<ModalClose onClick={onClose} />
			</ModalHeader>
			<ModalContent>
				<Box display='flex' flexDirection='column' mie={10} mbe={24} flexGrow={1}>
					<Label>{t('Format')}</Label>
					<Field>
						<FieldRow>
							<FieldLabel htmlFor='JSONField'>{t('JSON')}</FieldLabel>
							<Controller
								name='type'
								control={control}
								render={({ field: { value, onChange, ...field } }) => (
									<RadioButton
										id='json'
										{...field}
										onChange={() => onChange('json')}
										aria-describedby='JSONField'
										checked={value === 'json'}
									/>
								)}
							/>
						</FieldRow>
					</Field>
					<Field>
						<FieldRow>
							<FieldLabel htmlFor='plainTextField'>{t('CSV')}</FieldLabel>
							<Controller
								name='type'
								control={control}
								render={({ field: { value, onChange, ...field } }) => (
									<RadioButton
										id='csv'
										{...field}
										onChange={() => onChange('csv')}
										aria-describedby='plainTextField'
										checked={value === 'csv'}
									/>
								)}
							/>
						</FieldRow>
					</Field>
				</Box>
				<Box display='flex' flexDirection='column' mie={10} mbe={24} flexGrow={1}>
					<Label>{t('Export_most_recent_logs')}</Label>
					<Field>
						<FieldRow>
							<FieldLabel htmlFor='maxLogsExport'>{t('Max_logs_export')}</FieldLabel>
							<Controller
								name='count'
								control={control}
								render={({ field: { value, onChange, ...field } }) => (
									<RadioButton
										id='max'
										{...field}
										onChange={() => onChange('max')}
										aria-describedby='maxLogsExport'
										checked={value === 'max'}
									/>
								)}
							/>
						</FieldRow>
					</Field>
					<Field>
						<FieldRow>
							<FieldLabel htmlFor='customMaxLogs'>{t('Limit_number_of_logs')}</FieldLabel>
							<Controller
								name='count'
								control={control}
								render={({ field: { value, onChange, ...field } }) => (
									<RadioButton
										id='custom'
										{...field}
										onChange={() => onChange('custom')}
										aria-describedby='customMaxLogs'
										checked={value === 'custom'}
									/>
								)}
							/>
						</FieldRow>
						<Controller
							name='customExportAmount'
							control={control}
							rules={{ required: count === 'custom', min: 1, max: 2000 }}
							render={({ field }) => (
								<NumberInput
									id='limit'
									min={1}
									max={2000}
									required={count === 'custom'}
									disabled={count !== 'custom'}
									placeholder='100'
									{...field}
									aria-describedby='limitNumber'
								/>
							)}
						/>
					</Field>
				</Box>
			</ModalContent>
			<ModalFooter>
				<ModalFooterControllers>
					<Button onClick={onClose}>{t('Cancel')}</Button>
					<Button primary disabled={!isValid} onClick={handleConfirm}>
						{t('Download')}
					</Button>
				</ModalFooterControllers>
			</ModalFooter>
		</Modal>
	);
};
