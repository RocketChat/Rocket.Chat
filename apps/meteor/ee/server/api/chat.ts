import type { IMessage, ReadReceipt } from '@rocket.chat/core-typings';
import { License } from '@rocket.chat/license';
import { Meteor } from 'meteor/meteor';

import { API } from '../../../app/api/server/api';
import { getReadReceiptsFunction } from '../methods/getReadReceipts';

type GetMessageReadReceiptsProps = {
	messageId: IMessage['_id'];
};

declare module '@rocket.chat/rest-typings' {
	// eslint-disable-next-line @typescript-eslint/naming-convention
	interface Endpoints {
		'/v1/chat.getMessageReadReceipts': {
			GET: (params: GetMessageReadReceiptsProps) => {
				receipts: ReadReceipt[];
			};
		};
	}
}

API.v1.addRoute(
	'chat.getMessageReadReceipts',
	{ authRequired: true },
	{
		async get() {
			if (!License.hasModule('message-read-receipt')) {
				throw new Meteor.Error('error-action-not-allowed', 'This is an enterprise feature');
			}

			const { messageId } = this.queryParams;
			if (!messageId) {
				return API.v1.failure({
					error: "The required 'messageId' param is missing.",
				});
			}

			return API.v1.success({
				receipts: await getReadReceiptsFunction(messageId, this.userId),
			});
		},
	},
);
