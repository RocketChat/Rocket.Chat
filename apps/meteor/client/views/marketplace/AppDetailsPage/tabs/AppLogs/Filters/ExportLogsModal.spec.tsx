import { mockAppRoot } from '@rocket.chat/mock-providers';
import { composeStories } from '@storybook/react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';

import * as stories from './ExportLogsModal.stories';

const testCases = Object.values(composeStories(stories)).map((Story) => [Story.storyName || 'Story', Story]);
const onConfirm = jest.fn();
const { Default } = composeStories(stories);

afterEach(() => {
	jest.clearAllMocks();
});

test.each(testCases)(`renders without crashing`, async (_storyname, Story) => {
	const view = render(<Story />, {
		wrapper: mockAppRoot().build(),
	});
	expect(view.baseElement).toMatchSnapshot();
});

test.each(testCases)('Should have no a11y violations', async (_storyname, Story) => {
	const { container } = render(<Story />, { wrapper: mockAppRoot().build() });

	const results = await axe(container);

	expect(results).toHaveNoViolations();
});

it('should send the correct payload to the endpoint', async () => {
	render(<Default onConfirm={onConfirm} />, {
		wrapper: mockAppRoot().build(),
	});

	expect(screen.getByRole('button', { name: 'Download' })).toBeInTheDocument();
	await userEvent.click(screen.getByRole('button', { name: 'Download' }));
	expect(onConfirm).toHaveBeenCalledTimes(1);
	expect(onConfirm).toHaveBeenCalledWith('/api/apps/undefined/export-logs?count=2000&type=json');
});

it('should include instance filter in the payload to endpoint', async () => {
	render(
		<Default
			onConfirm={onConfirm}
			filterValues={{
				instance: '123',
			}}
		/>,
		{
			wrapper: mockAppRoot().build(),
		},
	);

	expect(screen.getByRole('button', { name: 'Download' })).toBeInTheDocument();
	await userEvent.click(screen.getByRole('button', { name: 'Download' }));
	expect(onConfirm).toHaveBeenCalledTimes(1);
	expect(onConfirm).toHaveBeenCalledWith('/api/apps/undefined/export-logs?instanceId=123&count=2000&type=json');
});
