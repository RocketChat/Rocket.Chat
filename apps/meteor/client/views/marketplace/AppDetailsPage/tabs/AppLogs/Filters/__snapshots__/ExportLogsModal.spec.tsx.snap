// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-css-4jy2w6"
    >
      <dialog
        aria-modal="true"
        class="rcx-box rcx-box--full rcx-modal"
        open=""
      >
        <div
          class="rcx-box rcx-box--full rcx-modal__inner rcx-css-1e2ego0"
        >
          <header
            class="rcx-box rcx-box--full rcx-modal__header"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__header-inner"
            >
              <h2
                class="rcx-box rcx-box--full rcx-modal__title rcx-css-trljwa rcx-css-lma364"
              >
                Export
              </h2>
              <button
                aria-label="Close"
                class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button rcx-css-trljwa rcx-css-lma364"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-icon--name-cross rcx-icon rcx-css-4pvxx3"
                >
                  î€ž
                </i>
              </button>
            </div>
          </header>
          <div
            class="rcx-box rcx-box--full rcx-modal__content rcx-css-1vw7itl"
          >
            <div
              class="rcx-box rcx-box--full rcx-modal__content-wrapper rcx-css-56vo9k"
            >
              <div
                class="rcx-box rcx-box--full rcx-css-6ua2r5"
              >
                <label
                  class="rcx-box rcx-box--full rcx-label"
                >
                  Format
                </label>
                <div
                  class="rcx-box rcx-box--full rcx-field"
                >
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <label
                      class="rcx-box rcx-box--full rcx-field__label rcx-label"
                      for="JSONField"
                    >
                      JSON
                    </label>
                    <label
                      class="rcx-box rcx-box--full rcx-radio-button"
                    >
                      <input
                        aria-describedby="JSONField"
                        checked=""
                        class="rcx-box rcx-box--full rcx-radio-button__input"
                        id="json"
                        name="type"
                        type="radio"
                      />
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-radio-button__fake"
                      />
                    </label>
                  </span>
                </div>
                <div
                  class="rcx-box rcx-box--full rcx-field"
                >
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <label
                      class="rcx-box rcx-box--full rcx-field__label rcx-label"
                      for="plainTextField"
                    >
                      CSV
                    </label>
                    <label
                      class="rcx-box rcx-box--full rcx-radio-button"
                    >
                      <input
                        aria-describedby="plainTextField"
                        class="rcx-box rcx-box--full rcx-radio-button__input"
                        id="csv"
                        name="type"
                        type="radio"
                      />
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-radio-button__fake"
                      />
                    </label>
                  </span>
                </div>
              </div>
              <div
                class="rcx-box rcx-box--full rcx-css-6ua2r5"
              >
                <label
                  class="rcx-box rcx-box--full rcx-label"
                >
                  Export_most_recent_logs
                </label>
                <div
                  class="rcx-box rcx-box--full rcx-field"
                >
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <label
                      class="rcx-box rcx-box--full rcx-field__label rcx-label"
                      for="maxLogsExport"
                    >
                      Max_logs_export
                    </label>
                    <label
                      class="rcx-box rcx-box--full rcx-radio-button"
                    >
                      <input
                        aria-describedby="maxLogsExport"
                        checked=""
                        class="rcx-box rcx-box--full rcx-radio-button__input"
                        id="max"
                        name="count"
                        type="radio"
                      />
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-radio-button__fake"
                      />
                    </label>
                  </span>
                </div>
                <div
                  class="rcx-box rcx-box--full rcx-field"
                >
                  <span
                    class="rcx-box rcx-box--full rcx-field__row"
                  >
                    <label
                      class="rcx-box rcx-box--full rcx-field__label rcx-label"
                      for="customMaxLogs"
                    >
                      Limit_number_of_logs
                    </label>
                    <label
                      class="rcx-box rcx-box--full rcx-radio-button"
                    >
                      <input
                        aria-describedby="customMaxLogs"
                        class="rcx-box rcx-box--full rcx-radio-button__input"
                        id="custom"
                        name="count"
                        type="radio"
                      />
                      <i
                        aria-hidden="true"
                        class="rcx-box rcx-box--full rcx-radio-button__fake"
                      />
                    </label>
                  </span>
                  <input
                    aria-describedby="limitNumber"
                    class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-number rcx-input-box"
                    disabled=""
                    id="limit"
                    max="2000"
                    min="1"
                    name="customExportAmount"
                    placeholder="100"
                    size="1"
                    type="number"
                    value="100"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rcx-box rcx-box--full rcx-modal__footer rcx-css-17mu816"
          >
            <div
              class="rcx-button-group rcx-button-group--align-end"
              role="group"
            >
              <button
                class="rcx-box rcx-box--full rcx-button rcx-button-group__item"
                type="button"
              >
                <span
                  class="rcx-button--content"
                >
                  Cancel
                </span>
              </button>
              <button
                class="rcx-box rcx-box--full rcx-button--primary rcx-button rcx-button-group__item"
                disabled=""
                type="button"
              >
                <span
                  class="rcx-button--content"
                >
                  Download
                </span>
              </button>
            </div>
          </div>
        </div>
      </dialog>
    </div>
  </div>
</body>
`;
